yarn-project/prover-client/src/orchestrator/orchestrator.ts

```diff
@@ -392,6 +396,55 @@ export class ProvingOrchestrator implements BlockProver {
     this.provingState?.cancel();
   }
 
+  /**
+   * Extract the block header from public inputs.
+   * TODO(#7346): Refactor this once new batch rollup circuits are integrated
+   * @returns The header of this proving state's block.
+   */
+  private async extractBlockHeader() {
+    if (
+      !this.provingState ||
+      !this.provingState.blockRootRollupPublicInputs ||
+      !this.provingState.finalRootParityInput?.publicInputs.shaRoot
+    ) {
+      throw new Error(`Invalid proving state, a block must be proven before its header can be extracted.`);
+    }
+
+    const rootRollupOutputs = this.provingState.blockRootRollupPublicInputs;
+    const previousMergeData = this.provingState.getMergeInputs(0).inputs;
+
+    if (!previousMergeData[0] || !previousMergeData[1]) {
+      throw new Error(`Invalid proving state, final merge inputs before block root circuit missing.`);
+    }
+
+    const contentCommitment = new ContentCommitment(
+      new Fr(previousMergeData[0].numTxs + previousMergeData[1].numTxs),
+      sha256Trunc(
+        Buffer.concat([
+          previousMergeData[0]?.txsEffectsHash.toBuffer(),
+          previousMergeData[1]?.txsEffectsHash.toBuffer(),
```

Comment:
Hm, do we need optional chaining here? The compiler should be able to infer that the merge data objects are defined based on the `if` statement above.

Reply:
Good point! Think it was leftover from copying to a new fn. Will remove the `?`s